README.md

```diff
@@ -50,6 +50,34 @@ Documentation for older versions or branches of s2n-tls can be generated locally
 
 Doxygen installation instructions are available at the [Doxygen](https://doxygen.nl/download.html) webpage.
 
+## Platform Support
+
+We’ve listed the distributions and platforms under two tiers: Tier 1 platforms are guaranteed to build, run, and pass tests in CI. Tier 2 platforms are guaranteed to build and we'll address issues opened against them, but they aren't currently running in our CI and are not actively reviewed with every commit. If you use a platform not listed below and would like to request (or help!) add it to our CI, please open an issue for discussion.
+
+### Tier 1
+
+|Distribution in CI	                                    |Platforms |
+|-------------------------------------------------------|---------|
+|Ubuntu18/22/24**                                       | x86_64  |
+|Ubuntu22                                               | i686    |
+|AL2, AL2023**                                          | x86_64  |
+|NixOS                                                  | x86_64, aarch64 |
+|OpenBSD [7.4](https://github.com/cross-platform-actions/action/blob/master/readme.md#supported-platforms)| x86_64  |
+|FreeBSD [latest](https://github.com/vmactions/freebsd-vm/blob/v1/conf/default.release.conf)| x86_64  |
+|OSX [latest](https://github.com/actions/runner-images?tab=readme-ov-file#available-images) | aarch64 |
+
+**Work in Progress
+
+### Tier 2
+
+|Distribution not in CI	|Platforms|
+|-----------------------|--------|
+| Fedora Core 34-36     |x86_64, aarch64 |
+| Ubuntu14/16/20        |x86_64, aarch64 |
+| OSX	                |x86_64|
+
```

Comment:
Should we put a lower bound on this one? As we discussed in [this PR](https://github.com/aws/s2n-tls/pull/4674), we don't really plan to support versions of OSX that even apple doesn't support anymore.